src/config/loading.rs

```diff
@@ -64,20 +74,29 @@ pub fn process_paths(config_paths: &[(PathBuf, FormatHint)]) -> Option<Vec<(Path
             std::process::exit(exitcode::CONFIG);
         }
 
-        for path in matches {
-            paths.push((path, *format));
+        match config_path {
+            ConfigPath::File(_, format) => {
+                for path in matches {
+                    paths.push(ConfigPath::File(path, *format));
+                }
+            }
+            ConfigPath::Dir(_) => {
+                for path in matches {
+                    paths.push(ConfigPath::Dir(path))
+                }
+            }
         }
     }
 
-    paths.sort_by(|(a, _), (b, _)| a.cmp(b));
+    paths.sort();
```

Comment:
I'm curious, how much does ordering matter for the config files?

Reply:
Yeah, it isn't clear to me why we were sorting here; I just decided to maintain that behavior though.